src/builder.ts

```diff
@@ -565,25 +594,98 @@ export class SchemaBuilder {
       return true
     }
 
-    const message = `${typeName}.${fieldName} is referencing a type "${outputType}" that is not defined in your GraphQL schema.`
+    const message = `Your GraphQL \`${typeName}\` object definition is projecting a field \`${fieldName}\` with \`${outputType}\` as output type, but \`${outputType}\` is not defined in your GraphQL Schema`
 
-    if (stage === 'build') {
-      if (isDevMode()) {
-        if (this.options.onUnknownFieldType) {
-          this.options.onUnknownFieldType({
-            unknownFieldType: outputType,
-            typeName,
-            fieldName,
-            error: new Error(message),
-          })
-        } else {
-          console.log(`Warning: ${message}`)
-        }
-      } else {
-        throw new Error(message)
-      }
-    }
+    registerHook(
+      this.options.onUnknownFieldType,
+      [
+        {
+          unknownFieldType: outputType,
+          typeName,
+          fieldName,
+          error: new Error(message),
+        },
+      ],
+      message,
+      stage,
+    )
 
     return false
   }
+
+  assertArgNameExists(
+    parentTypeName: string,
+    prismaOutputTypeName: string,
+    fieldName: string,
+    config: FieldPublisherConfig,
+    stage: 'build' | 'walk',
+    configProperty: 'filtering' | 'ordering',
+  ): { wrongArgNames: string[] } | true {
+    if (!config[configProperty] || config[configProperty] === true) {
+      return true
+    }
+
+    const argNames = Object.keys(
+      config[configProperty] as Record<string, boolean>,
```

Comment:
I would prefer `as Index<Boolean>` but I know TimG didn't like that, can't remember what you and I concluded between us in the past about it.